### PR TITLE
VB-9: add redis service

### PR DESCRIPTION
**Pull request by voonya**

### CODE REVIEW

File: packages/backend/src/services/redis-storage.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+﻿import type { createClient } from 'redis';
+
+export type RedisClientType = ReturnType<typeof createClient>;
+
+export class RedisStorageService {
+  private client: RedisClientType;
+
+  constructor(redisClient: RedisClientType) {
+    this.client = redisClient;
```
Comment (Nightlngale) [inline packages/backend/src/services/redis-storage.ts:9]: You don't need to pass redisClient. Just instantiate it inside redisStorageService

File: packages/backend/src/services/redis-storage.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  }
+
+  async get(
+    key: string,
+  ): Promise<string | Array<any> | Record<string, unknown>> {
```
Comment (Nightlngale) [inline packages/backend/src/services/redis-storage.ts:14]: do not use any

File: packages/backend/src/services/redis-storage.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      return null;
+    }
+
+    const data = await this.client.get(key);
+    try {
+      return JSON.parse(data);
+    } catch {
+      return data;
+    }
+  }
```
Comment (Nightlngale) [inline packages/backend/src/services/redis-storage.ts:26]: You can always store the data as a JSON. I that can you can simplify this step

**Review by Nightlngale — CHANGES_REQUESTED** [review]

File: packages/backend/src/services/redis-storage.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    const data = await this.client.get(key);
+
+    return JSON.parse(data);
+  }
+
+  // expirationTime in milliseconds
+  async set(
+    key: string,
+    data: RedisStorageSetDataType,
+    expirationTime?: number,
```
Comment (Vitalii4as) [inline packages/backend/src/services/redis-storage.ts:52]: just rename `expirationTimeMs` and no comment needed anymore

Comment (voonya) [reply packages/backend/src/services/redis-storage.ts]: fixed

File: packages/backend/src/services/redis-storage.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    this.client = createClient(redisConnectionParams);
+
+    this.client.on('error', (err) => console.log('redis client error', err));
+
+    this.client.connect().then(() => {
+      console.log('redis client connected!');
+    });
+  }
+
+  async get(key: string): Promise<RedisStorageGetDataType> {
```
[the remark this answers is not in the document]
Comment (voonya) [reply packages/backend/src/services/redis-storage.ts]: fixed

File: packages/backend/src/helpers/env.ts [hunk 6]
```diff
@@ -1,3 +1,8 @@
-type ProcessEnvKey = 'NODE_ENV' | 'PORT';
+type ProcessEnvKey =
+  | 'NODE_ENV'
+  | 'PORT'
+  | 'REDIS_PORT'
```
Comment (dmitriy-uvin) [inline packages/backend/src/helpers/env.ts:4]: create another type for redis envs, for example RedisEnvKey

**Comment by dmitriy-uvin** [comment]
Instead of union in function directly, we can use type EnvKey = Redis | Process, and then in function (key: EnvKey)

**Review by Nightlngale — APPROVED** [review]

**Pull request merged**